src/System.Text.Encoding.CodePages/tests/EncoderFallbackBufferHelper.cs

```diff
@@ -0,0 +1,37 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Xunit;
+
+namespace System.Text.Tests
+{
+    public partial class EncoderFallbackBufferHelperTest : IClassFixture<CultureSetup>
+    {
+        public EncoderFallbackBufferHelperTest(CultureSetup setup)
+        {
+            // Setting up the culture happens externally, and only once, which is what we want.
+            // xUnit will keep track of it, do nothing.
+        }
+
+        [Fact]
+        public static void TestEncoderFallbackBufferHelper()
+        {
+            // Add the code page provider.
+            Encoding.RegisterProvider(CodePagesEncodingProvider.Instance);
+
+            // Use windows-1252 with a custom fallback that utilizes the EncoderFallbackBufferHelper.
+            Encoding encoding = Encoding.GetEncoding(1252, new TestEncoderFallback(), DecoderFallback.ReplacementFallback);
+
+            // Roundtrip unsupported character.
+            byte[] bytes = encoding.GetBytes("\u0661");
+            char[] chars = encoding.GetChars(bytes);
+            Assert.Equal(new string(chars), "?");
```

Comment:
other way around, `Assert.Equal(expected, actual)` not `Assert.Equal(actual, expected)`

Reply:
👍